docs/api-reference.md

```diff
@@ -3,25 +3,623 @@ id: api-reference
-    "type": "createShoppingList",
-    "payload": {
-        "name": "List 1"
-    }
+  "aggregateName":"Todo",
+  "type":"createItem",
+  "aggregateId":"root-id",
+  "payload": {
+    "id":`date +%s`,
+    "text":"Learn reSolve API"
+  }
 }
 '
 ```
+
+### Resolve-Redux Library
+
+The reSolve framework includes the client **resolve-redux** library used to connect a client React + Redux app to a reSolve-powered backend. This library provides the following HOCs:
+
+| Function Name                                     | Description                                                                                                      |
+| ------------------------------------------------- | ---------------------------------------------------------------------------------------------------------------- |
+| [connectViewModel](#connectviewmodel)             | Connects a React component to a reSolve View Model.                                                              |
+| [connectReadModel](#connectreadmodel)             | Connects a React component to a reSolve Read Model.                                                              |
+| [connectRootBasedUrls](#connectrootbasedurls)     | Fixes URLs passed to the specified props so that they take into respect the correct root folder path.            |
+| [connectStaticBasedUrls](#connectstaticbasedurls) | Fixes URLs passed to the specified props so that they take into respect the correct static resource folder path. |
+
+#### connectViewModel
+
+Connects a React component to a reSolve View Model.
+
+##### Example
+
+```js
+export const mapStateToOptions = (state, ownProps) => {
+  const aggregateId = ownProps.match.params.id
+
+  return {
+    viewModelName: 'ShoppingList',
+    aggregateIds: [aggregateId]
+  }
+}
+
+export const mapStateToProps = (state, ownProps) => {
+  const aggregateId = ownProps.match.params.id
+
+  return {
+    aggregateId
+  }
+}
+
+export const mapDispatchToProps = (dispatch, { aggregateActions }) =>
+  bindActionCreators(
+    {
+      ...aggregateActions,
+      replaceUrl: routerActions.replace
+    },
+    dispatch
+  )
+
+export default connectViewModel(mapStateToOptions)(
+  connect(
+    mapStateToProps,
+    mapDispatchToProps
+  )(ShoppingList)
+)
+```
+
+#### connectReadModel
+
+Connects a React component to a reSolve Read Model.
+
+##### Example
+
+```js
+export const mapStateToOptions = () => ({
+  readModelName: 'ShoppingLists',
+  resolverName: 'all',
+  resolverArgs: {}
+})
+
+export const mapStateToProps = (state, ownProps) => ({
+  lists: ownProps.data
+})
+
+export const mapDispatchToProps = (dispatch, { aggregateActions }) =>
+  bindActionCreators(aggregateActions, dispatch)
+
+export default connectReadModel(mapStateToOptions)(
+  connect(
+    mapStateToProps,
+    mapDispatchToProps
+  )(MyLists)
+)
+```
+
+#### connectRootBasedUrls
+
+Fixes URLs passed to the specified props so that they take into respect the correct root folder path.
```

Comment:
"take into respect"? Where did you find this phrase?

Reply:
No answer?

Reply:
Why resubmit this PR without answering all the questions or making all the necessary corrections?